### PR TITLE
Minimum cost to construct temple in C++

### PR DESCRIPTION
**Pull request by gayatri517**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Code added in cp folder.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#7235

### CODE REVIEW

**Review by bhav09 — COMMENTED** [review]
The code is not comented. Please comment the code.  Make sure the comments are above the line of code and not alongside.

**Comment by gayatri517** [comment]
@bhav09 i have added the comments. Kindly review it

File: C-Plus-Plus/cp/min_cost_to_construct_temple.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        visited[(int)src]=true;
+        int count=1;
+        for(auto i : m[(int)src])
+        {
+            if(!visited[i])
+            {
+              count+= dfshelper(i,visited);
+            }
+        }
+    return count;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/min_cost_to_construct_temple.cpp]: Add a one tab space

File: C-Plus-Plus/cp/min_cost_to_construct_temple.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            then make the road */
+            if(!visited[(int)i.first])
+            {
+                count=dfshelper(i.first,visited);
+                count--;
+                sum+=(count*cost_road + cost_temple);
+
+            }
+        }
+    return sum;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/min_cost_to_construct_temple.cpp]: Add a one tab space

File: C-Plus-Plus/cp/min_cost_to_construct_temple.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        g.add(p,q);
+    }
+    /* If the cost of road is not less than the cost of temple,
+    then construct the temple */
+    if(cost_road>cost_temple) 
+    {
+        sum=cost_temple*vertices;
+    }
+    else
+    sum=g.dfs();
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/min_cost_to_construct_temple.cpp]: Add a one tab space

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by gayatri517** [comment]
@ankitaggarwal23 I have done the changes.
Kindly review it

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**